sdk/iot/pnp/tests/cmocka/test_az_iot_pnp_client.c

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_pnp_client.h"
+#include <az_iot_pnp_client.h>
+#include <az_span.h>
+
+#include <az_precondition.h>
+#include <az_precondition_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <az_test_precondition.h>
+#include <cmocka.h>
+
+// PnP Client
+#define TEST_DEVICE_ID_STR "my_device"
+#define TEST_DEVICE_HOSTNAME_STR "myiothub.azure-devices.net"
+#define TEST_MODULE_ID "my_module_id"
+#define TEST_ROOT_INTERFACE_NAME "my_root_interface_name"
+#define TEST_USER_AGENT "os=azrtos"
+#define TEST_CONTENT_TYPE "my_content_type"
+#define TEST_CONTENT_ENCODING "my_content_encoding"
+
+static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID_STR);
+static const az_span test_device_hostname = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_HOSTNAME_STR);
+static const az_span test_root_interface_name = AZ_SPAN_LITERAL_FROM_STR(TEST_ROOT_INTERFACE_NAME);
+static const az_span test_user_agent = AZ_SPAN_LITERAL_FROM_STR(TEST_USER_AGENT);
+static const az_span test_content_type = AZ_SPAN_LITERAL_FROM_STR(TEST_CONTENT_TYPE);
+static const az_span test_content_encoding = AZ_SPAN_LITERAL_FROM_STR(TEST_CONTENT_ENCODING);
+
+static const char test_correct_pnp_user_name[]
+    = "myiothub.azure-devices.net/my_device/"
+      "?api-version=2018-06-30&digital-twin-model-id=" TEST_ROOT_INTERFACE_NAME;
+
+static const char test_correct_pnp_user_name_with_user_agent[]
+    = "myiothub.azure-devices.net/my_device/?api-version=2018-06-30&" TEST_USER_AGENT
+      "&digital-twin-model-id=" TEST_ROOT_INTERFACE_NAME;
+
+#ifndef NO_PRECONDITION_CHECKING
+
+enable_precondition_check_tests()
+
+    static void test_az_iot_pnp_client_init_NULL_client_fails(void** state)
```

Comment:
I'd like to see if there's a way we can fix this. I've ran into this issue making my own tests and the clang format doesn't like that `enable_precondition_check_tests()` doesn't have a semicolon so the next function gets indented too much. Not for this PR but I'd like to hopefully find a solution for it.

Reply:
Any reason we can't just stick a semicolon here :)?  I tried it and clang doesn't do the indent thing (good catch btw).

I have seen certain extra-extra-strict compilations from gcc, where if there was like an "int foo;;" then it'd complain on semicolon number 2.  So maybe that'd discourage an unneeded ; after the "{...}" block the macro expands to??

FWIW I'm not going to hold checkin on this, but will put in as-is since it's a bigger problem you're tracking.